tests/test_utils.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import utils
+import unittest
+
+
+class TestUtils(unittest.TestCase):
+    def test_append_unless(self):
+        # unless = None
+        self.assertEqual('address:port', utils.append_unless(None, 'address', ':port'))
+        self.assertEqual(None, utils.append_unless(None, None, ':port'))
+
+        # unless = string
+        self.assertEqual('basesuffix', utils.append_unless('bad', 'base', 'suffix'))
+        self.assertEqual('bad', utils.append_unless('bad', 'bad', 'suffix'))
+
+        # unless = None, args = lists
+        self.assertEqual([1, 2, 3], utils.append_unless(None, [1, 2], [3]))
+        self.assertEqual(None, utils.append_unless(None, None, [3]))
+
+    def test_md5(self):
+        self.assertEqual('50e1eeff19b2501c791612cd907fff7a',
+                         utils.md5('test string\nwith newline'))
+
+    def test_fetch_url(self):
```

Comment:
Aren't you effectively testing a third party library, given your `fetch_url` is a very lightweight wrapper around one of its methods. Is it worth this much effort :-) .

Reply:
The purpose of the test is to test the interface for `fetch_url`, and to that end I could have used `with patch(...)` instead. Thanks for pointing out.

Reply:
`with patch(...)` sounds simpler.